onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -158,10 +163,7 @@ void MIGraphXExecutionProvider::get_flags_from_session_info(const MIGraphXExecut
   }
 
   // Save/load migraphx compiled models
-  save_compiled_model_ = info.save_compiled_model;
-  save_compiled_path_ = info.save_model_file;
-  load_compiled_model_ = info.load_compiled_model;
-  load_compiled_path_ = info.load_model_file;
+  model_cache_path_ = info.model_cache_dir;
```

Comment:
While you are welcomed to use std::filesystem::path, every time when you assign a value to this type you need carefully check the source string's encoding. 
On Windows we expect the source string is in native code page encoding. Otherwise if the string is encoded as UTF-8 you must manually convert it to std::wstring before assign it to a path.  
Or you may change type of model_cache_dir to const ORTCHAR_T*.

Reply:
Odd, these came from windows side changes and we haven't seen this effecting our runs.

Reply:
So, if you have a std::wstring or ORT_CHART* string, you have no problem. Otherwise try to explicitly state the encoding. Otherwise people will assume the string is a UTF-8 string.